lib/avromatic/model/types/fixed_type.rb

```diff
@@ -0,0 +1,37 @@
+module Avromatic
+  module Model
+    module Types
+      class FixedType
+        VALUE_CLASSES = [::String].freeze
+
+        attr_reader :size
+
+        def initialize(size)
+          @size = size
+        end
+
+        def value_classes
+          VALUE_CLASSES
+        end
+
+        def coerce(input)
+          if coercible?(input)
+            input
+          else
+            raise ArgumentError.new("Could not coerce '#{input.inspect}' to a Fixed(#{size})")
```

Comment:
What do you think about defining a `CoercionError` class within this gem instead of using `ArgumentError`?

Maybe that's already covered by the improved coercion errors that you have in mind.